spras/analysis/summary.py

```diff
@@ -33,34 +34,38 @@ def summarize_networks(file_paths: Iterable[Path], node_table: pd.DataFrame, alg
         nodes_by_col.append(set(node_table.loc[node_table[col] > 0, 'NODEID']))
 
     # Initialize list to store network summary data
-    nw_info = []
+    nw_info: list[list[Any]] = []
 
     algo_with_params = sorted(algo_with_params)
 
     # Iterate through each network file path
     for index, file_path in enumerate(sorted(file_paths)):
         with open(file_path, 'r') as f:
-            lines = f.readlines()[1:]  # skip the header line
+            lines = f.readlines()[1:] # as this is an output edge file, we skip the header line
 
         # directed or mixed graphs are parsed and summarized as an undirected graph
         nw = nx.read_edgelist(lines, data=(('weight', float), ('Direction', str)))
 
-        # Save the network name, number of nodes, number edges, and number of connected components
-        nw_name = str(file_path)
-        number_nodes = nw.number_of_nodes()
-        number_edges = nw.number_of_edges()
-        ncc = nx.number_connected_components(nw)
+        # Begin collecting data into a dictionary.
+        # (We use dictionaries over list to make )
+        data: OrderedDict[str, Any] = OrderedDict()
+
+        data.update([
+            ('Name', str(file_path)),
+            ('Number of nodes', nw.number_of_nodes()),
+            ('Number of edges', nw.number_of_edges()),
+            ('Number of connected components', nx.number_connected_components(nw)),
+        ])
 
         # Save the max/median degree, average clustering coefficient, and density
-        if number_nodes == 0:
-            max_degree = 0
-            median_degree = 0.0
-            density = 0.0
-        else:
-            degrees = [deg for _, deg in nw.degree()]
-            max_degree = max(degrees)
-            median_degree = median(degrees)
-            density = nx.density(nw)
+        degrees = [deg for _, deg in nw.degree()]
+        data.update([
+            # If the number of nodes are 0, degrees will equal [],
```

Comment:
Do we have a test case covering that or did you manually verify that behavior?

Reply:
I manually verified that behavior - I thought it was part of the `Graph#degree` contract [but it's not](https://networkx.org/documentation/stable/reference/classes/generated/networkx.Graph.degree.html)? Either way is fine, but switching back to the quick `if` check here is very reasonable given the above info.